onnxruntime/test/optimizer/qdq_test_utils.h

```diff
@@ -110,5 +127,65 @@ GetQDQTestCaseFn BuildQDQResizeTestCase(const std::vector<int64_t>& input_shape,
                                         const std::string& mode = "nearest",
                                         const std::string& coordinate_transformation_mode = "half_pixel");
 
+template <typename Input1Type, typename Input2Type, typename OutputType>
+GetQDQTestCaseFn BuildBinaryOpTestCase(const std::vector<int64_t>& input_shape,
+                                       const std::string& op_type) {
+  return [input_shape, op_type](ModelTestBuilder& builder) {
+    auto* input1_arg = builder.MakeInput<float>(input_shape, -1.f, 1.f);
+    auto* input2_arg = builder.MakeInput<float>(input_shape, -1.f, 1.f);
+    auto* output_arg = builder.MakeOutput();
+
+#ifdef USE_NNAPI  // NNAPI require consistent scales for DQ -> bin op -> Q
+    float q_scale = 0.008f;
```

Comment:
Is this the case for all binary ops? Naively I would have expected mul or div to potentially change the scale but maybe that isn't the case.

Reply:
Here's a snippet from a QDQ model exported from pytorch where the scale and zero point changes across an Add node.

<img width="157" alt="image" src="https://user-images.githubusercontent.com/979079/152925348-a2ae930c-06e4-4c8e-8eee-1d580a0800da.png">

Reply:
Yes, the output scale and ZP can be changed, updated the test case, it is only required to be the same for QD->bin_op_input and bin_op_output->Q